phac-nml/irida-next
# Data Export - Destroy Service

## What does this PR do and why?
This PR adds the `DataExports` `DestroyService`. The service will eventually be called from the `data_exports_controller` and handles destroying the `DataExport` model instance as well as immediately purging the zip folder attached to the `DataExport`, if it still exists.

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

## How to set up and validate locally
1. In `rails c`, create a `data_export` as described in the [create service PR](https://github.com/phac-nml/irida-next/pull/444)
2. Assign the above export to variable `data_export = user.data_exports.last`
3. Delete the export with `DataExports::DestroyService.new(data_export, user).execute`
4. Ensure the export was deleted with `data_export.deleted?`

Test permission

5. Repeat the above steps, but assign a different user in the `DestroyService`
6. Ensure a permissions error occurs.

7. Ensure tests pass locally.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
